config/proto_conversions.go

```diff
@@ -650,19 +650,21 @@ func NetworkConfigToProto(network *NetworkConfig) (*pb.NetworkConfig, error) {
 	return &proto, nil
 }
 
-// MaintenanceConfigToProto converts MaintenanceConfig from the proto equivalent.
+// MaintenanceConfigToProto converts MaintenanceConfig to the proto equivalent. This function will swallow
+// any resource name conversion errors and put the full SensorName in the Name field of the ResourceName proto object.
 func MaintenanceConfigToProto(maintenanceConfig *MaintenanceConfig) (*pb.MaintenanceConfig, error) {
-	// convert from string to resource name
-	name, err := resource.NewFromString(maintenanceConfig.SensorName)
-	if err != nil {
-		return nil, err
-	}
-
 	proto := pb.MaintenanceConfig{
-		SensorName:            protoRdkUtils.ResourceNameToProto(name),
 		MaintenanceAllowedKey: maintenanceConfig.MaintenanceAllowedKey,
 	}
 
+	if maintenanceConfig.SensorName != "" {
+		name, err := resource.NewFromString(maintenanceConfig.SensorName)
+		if err != nil {
```

Comment:
Sorry what's the issue we're fixing here? That some times `NewFromString` would error on maintenance config sensor names? What was the error?

Reply:
if SensorName is empty, NewFromString would error as `string \"\" is not a fully qualified resource name`. But malformed names would run into the issue as well, so this aims to knowingly return something that's borked

Reply:
When would a maintenance config's sensor name be `""`?

Reply:
https://viam.atlassian.net/browse/RSDK-10112, if the user skipped that field or left it blank. in the viam-server, it will fail validation but we shouldn't fail on app-side

Reply:
Gotcha cool; my bad should've looked at ticket beforehand.